Longest_Substring_Without_Repeat.java

```diff
@@ -0,0 +1,44 @@
+package Week1_Longest_Substring_without_repeat;
+import java.util.Scanner;
+import org.junit.Assert;
+public class Longest_Substring_Without_Repeat {
+
+	public static void main(String[] args) {
+		// TODO Auto-generated method stub
+		Scanner s=new Scanner(System.in);
+		System.out.println("Enter your String");
+		String str=s.nextLine();
+		System.out.println(lengthOfLongestSubstring(str));
+
+	}
+    public static int lengthOfLongestSubstring(String s) {
+    	if(s==null) {
+    		System.out.println("You have entered a null String");

```

Comment:
Typically its better for a method like this to not print out to the console window. What if the user of this method doesn't show a console window or doesn't want their methods to be output? Returning -1 as you do below is usually sufficient, or you can throw an exception with a detailed message and rely on the caller to decide what to do with it, like output the message to the console or just ignore it